### ISSUE TITLE
Daily timeseries with uncertainties 

### ISSUE THREAD

**Issue by parisaSHAFIEI** [groops-devs/groops#36]
Hi Sebastian, 

I have processed the GNSS data of zim2 station for one year, i.e. 2020, in GROOPS and got the kinematic positions as well as the variance-covariance matrices for each DOY, exactly like the ones which were in the example scenario. After that, because I wanted to get the daily time series with the uncertainties for the station. Therefore, I worked with "TimeSeriesCreate" program to get these parameters for each DOY. However, unfortunately, I could not get them and as a result, I wrote a program in python to estimate the average x-y-z coordinates as well as the standard errors for the whole year in form of time series. 

So, I am just wondering if there was any program inside GROOPS which I can use to create this daily time series with uncertainties, i.e. a text file with 6 columns or something similar containing MJD, X, Y, Z and the associated standard deviations.

Thank you in advance and I am looking forward to hearing from you. 

Best regards, 
Parisa

**Comment by parisaSHAFIEI** [comment]
Hi Patrick, 

Thank you for your great help. 

According to what you have told me, I could create the time series by merging the kinematic positions and the variance-covariance info for each DOY. The columns are mjd, x, y, z, variance_x, variance_y and variance_z. 

Besides, I could also calculate the mean of these parameters based on using the "InstrumentArcCalculate" program. However, since I want to plot the standard errors for the whole year, I want to have those standard deviation parameters instead of variances. Although I wrote a program in Python to calculate and plot those parameters, I am just wondering how I can use the "sqrt" function in GROOPS to get those standard deviations for x,y and z from the associated variances. 

Thank you again for the time you spent on answering my question and I am looking forward to hearing from you.

Best regards, 
Parisa

**Comment by sestras** [comment]
Hi Parisa,

Just use _sqrt(dataX)_ instead of _dataX_ for the variance outColumns in InstrumentArcCalculate. In the [documentation of that program](https://groops-devs.github.io/groops/html/InstrumentArcCalculate.html) you can see that outColumn is of type _expression_, which means you can use any [supported mathematical expression](https://groops-devs.github.io/groops/html/general.parser.html) in that field.

Cheers,
Sebastian

**Comment by zhedumi** [comment]
Hi Parisa,

I made a slight error in the case that I forgot to use _sqrt(dataX)_ within the script I added above. As Sebastian has already elaborated on you just need todo _sqrt(dataX)_ on the variances. Here is my example script now actually with the columns having mjd, x, y, z, std. x, std. y, std. z : [Updated 99mergeEpochwisePosCov.xml.tar.gz](https://github.com/groops-devs/groops/files/8250417/99mergeEpochwisePosCov.xml.tar.gz)

Best regards,

Patrick

**Comment by parisaSHAFIEI** [comment]
Dear Sebastian and Patrick, 

Based on your feedback, we could get closer to our final target in Precise Point Positioning for IGS ground stations.

So, I used sqrt(dataX) in my program and could get the standard deviations for each epoch and finally get the final mean values for each DOY. The example script was really helpful. 

We tried the "InstrumentConcatenate" to create file containing the mean of x, y, z, stdx, stdy and stdz per day for the entire year. What we got is just rewriting of daily means in the output file. For better understanding I have provided you with the two of the merged files that I want to use as input to merge them into one file with separate rows. 

[mean_valuses_for_merged_pos_with_cov_2020-01-01.zim2.txt](https://github.com/groops-devs/groops/files/8252481/mean_valuses_for_merged_pos_with_cov_2020-01-01.zim2.txt)
[mean_valuses_for_merged_pos_with_cov_2020-01-02.zim2.txt](https://github.com/groops-devs/groops/files/8252482/mean_valuses_for_merged_pos_with_cov_2020-01-02.zim2.txt)

Thank you very much for your cooperation and great support. 

Best regards, 
Parisa

**Comment by zhedumi** [comment]
Hi Parisa,

If I understand you correct you want to merge the daily data, then compute the mean values of each data column and finally have a file containing only the mean values. To merge them you will be required to use [loops](https://groops-devs.github.io/groops/docs/html/general.loopsAndConditions.html) within the groops script.

One approach is to create daily mean values for the whole year then use [InstrumentConcatenate](https://groops-devs.github.io/groops/html/InstrumentConcatenate.html) in combinations with [loops](https://groops-devs.github.io/groops/docs/html/general.loopsAndConditions.html) to gather them into one file.

Or another approach is to use [InstrumentConcatenate](https://groops-devs.github.io/groops/html/InstrumentConcatenate.html) to combine the position/covariance daily results into one file respectively, "merge the files by [InstrumentArcCalculate](https://groops-devs.github.io/groops/docs/html/InstrumentArcCalculate.html), synchronize them into daily arcs by [InstrumentSynchronize](https://groops-devs.github.io/groops/docs/html/InstrumentSynchronize.html) and as last step compute the mean values with [InstrumentArcCalculate](https://groops-devs.github.io/groops/docs/html/InstrumentArcCalculate.html). 

For easier understanding I put both approaches into the example groops file:
[99mergeEpochwisePosCov.xml.tar.gz](https://github.com/groops-devs/groops/files/8253133/99mergeEpochwisePosCov.xml.tar.gz)

Best regards,
Patrick

**Comment by parisaSHAFIEI** [comment]
Hi Patrick and Sebastian,

Firstly, I would like to appreciate the time you spent on answering my question.

As a matter of fact, I went through the example groops configuration file and it was exactly what I wanted to do. I just kept the first approach and got the final time series for the whole year. 
Besides, by going through the configuration file, I found my mistake in using the "InstrumentConcatenate" program. In fact, I must have set loop in input file, i.e. loop=timeLoop, to put all the instrument files inside one file. So, finally I obtained the results for the whole year.

Thank you again for your great consideration. 

Best regards, 
Parisa
